### PR TITLE
fix: ins_bs_one counting inline virtual text as tabs

### PR DESCRIPTION
**Pull request by Sam-programs**
Problem: `ins_bs_one` uses `getvcol` to counts the amount of spaces deleted, However `getvcol` counts inline virtual text as a part of a character's length, making `ins_bs_one` think it deleted virtual text length spaces and tries to restore it back to `want_vcol` by addding spaces.
Solution:
Make `ins_bs_one` shift vcol by the actual length of the deleted characters.

~~Perhaps `ins_bs` can be refactored to use actual characters instead of virtual columns ?~~

### CODE REVIEW

**Comment by Sam-programs** [comment]
I couldn't find a way to get the size of tabs without using `getvcol`, So the [issue](https://github.com/neovim/neovim/issues/26388) still exists for tabs but not spaces.

**Comment by glepnir** [comment]
`curbuf->b_p_ts` is tabstop

File: test/functional/api/extmark_spec.lua [hunk 1]
```diff
@@ -834,6 +834,38 @@ describe('API/extmarks', function()
...
+    feed(':%d<cr>')
+    feed('0i<space><esc>')
+    set_extmark(ns, marks[1], 0, 0,{
+      virt_text_pos = 'inline',
+      ephemeral = false,
+      virt_text = { { '[[ Virtual Text ]]' } },
+    })
+    feed('a<bs><esc>')
+    eq({''},helpers.buf_lines(0))
+  end)
```
Comment (zeertzjq) [inline test/functional/api/extmark_spec.lua:867]: Where are the TABs in this test?

**Comment by zeertzjq** [comment]
Also I may have a different solution:
```diff
diff --git a/src/nvim/edit.c b/src/nvim/edit.c
index 8984a959d..bd25a77da 100644
--- a/src/nvim/edit.c
+++ b/src/nvim/edit.c
@@ -3861,6 +3861,14 @@ static bool ins_bs(int c, int mode, int *inserted_space_p)
         ins_bs_one(&vcol);
       }
 
+      if (curbuf->b_virt_text_inline > 0) {
+        chartabsize_T cts;
+        init_chartabsize_arg(&cts, curwin, curwin->w_cursor.lnum, curwin->w_cursor.col,
+                             get_cursor_line_ptr(), get_cursor_pos_ptr());
+        (void)win_lbr_chartabsize(&cts, NULL);
+        vcol += cts.cts_cur_text_width_left + cts.cts_cur_text_width_right;
+      }
+
       // insert extra spaces until we are at want_vcol
       while (vcol < want_vcol) {
         // Remember the first char we inserted
```

Maybe this is better when deleting multiple spaces and there is inline virtual text in the middle.

File: test/functional/api/extmark_spec.lua [hunk 2]
```diff
@@ -834,6 +834,38 @@ describe('API/extmarks', function()
     check_undo_redo(ns, marks[1], 0, 4, 0, 6)
   end)
 
+  it("tabs don't write spaces when deleted with inline virtual text", function()
+    -- ins_bs in edit.c
+    feed(':set shiftwidth=2<cr><esc>')
+    feed(':set softtabstop=2<cr><esc>')
+    feed(':set tabstop=8<cr><esc>')
+    feed(':%d<cr>')
+    feed('0i<tab><esc>')
```
Comment (zeertzjq) [inline test/functional/api/extmark_spec.lua:843]: This doesn't insert a TAB. It inserts two spaces because 'shiftwidth' and 'smarttab' are set.

Comment (Sam-programs) [reply test/functional/api/extmark_spec.lua:844]: > When on, a \<Tab\> in front of a line inserts blanks according to
	'shiftwidth'.
	
Are 'blanks' spaces?

**Pull request closed without merging**